spacy/pipeline/entity_linker.py

```diff
@@ -474,18 +474,40 @@ def predict(self, docs: Iterable[Doc]) -> List[str]:
 
                 # Looping through each entity in batch (TODO: rewrite)
                 for j, ent in enumerate(ent_batch):
-                    sent_index = sentences.index(ent.sent)
-                    assert sent_index >= 0
+                    assert hasattr(ent, "sents")
+                    sents = list(ent.sents)
+                    # Note: the last sentence associated with an sentence-crossing entity isn't complete. E. g. if you
+                    # have "Mahler's Symphony No. 8 was beautiful", the entity being "No. 8", ent.sents would be:
+                    #   1. "Mahler's Symphony No."
+                    #   2. "8"
+                    # whereas doc.sents would be:
+                    #   1. "Mahler's Symphony No."
+                    #   2. "8 was beautiful"
+                    # This makes it tricky to receive the last sentence by indexing doc.sents - hence we use an offset
+                    # to determine sent_indices[1].
```

Comment:
This looks like a bug in `span.ents` - the second/last sentence shouldn't be partial. It looks like there's a bug in the indexing near the end of the doc. If you would a sentence after "beautiful", you'll see that the last sentence will in fact be complete. This bug only shows up when the entity crosses boundaries into the very last sentence, in which case the last sentence is truncated to the end of the span.

Let me know if you want to look into solving this bug (in a separate PR) or whether you'd like someone else to look into it.

Reply:
Ah, ok. I'd want to look into this myself, if we're ok with it not happening right away (due to other EL stuff I'd like to take of first).

Reply:
The thing is that I don't want this PR to be unnecessarily complicated because it's working around a bug in another part of the code base, you know?

Reply:
Right, so you think it'd be better to address the sentence indexing bug before we implement the solution for the EL bug?

Reply:
Alternatively we can also write this PR assuming that the `sents` work correctly, but then we really shouldn't forget to follow up with the other fix.

Reply:
Yeah, it's better to fix the sentence indexing problem first then. I should have a look at https://github.com/explosion/spaCy/blob/master/spacy/tokens/span.pyx#L427 then, no?

Reply:
[PR](https://github.com/explosion/spaCy/pull/12405) with test and fix.

Reply:
Merged!